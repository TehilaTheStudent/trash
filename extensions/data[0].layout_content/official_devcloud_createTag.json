[
  {
    "componentName": "d-form",
    "id": 1665991638516,
    "category": "form",
    "props": {
      "layout": "vertical",
      "ref": "formRef"
    },
    "children": [
      {
        "componentName": "d-form-item",
        "id": 1668220990172,
        "category": "form",
        "props": {
          "label": "Tag Name",
          "field": "tag_name",
          "help-tips": "",
          "rules": [
            {
              "required": true,
              "message": "标签名不能为空",
              "trigger": [
                "blur",
                "change"
              ]
            },
            {
              "regex": "^.{0,64}$",
              "message": "标签名不能超过64个字符",
              "trigger": [
                "blur",
                "change"
              ]
            }
          ],
          "visible": {
            "type": "fixed",
            "conditions": [],
            "value": true
          },
          "disabled": false
        },
        "children": [
          {
            "componentName": "d-input",
            "id": 1668220990171,
            "props": {
              "inputType": "normal",
              "placeholder": "请输入",
              "isTrim": true,
              "autocomplete": "off",
              "disabled": {
                "type": "fixed",
                "conditions": [],
                "value": false
              }
            },
            "category": "form"
          }
        ]
      },
      {
        "componentName": "d-form-item",
        "id": 1676461032692,
        "category": "form",
        "props": {
          "label": "Repository",
          "field": "repo_name",
          "help-tips": "",
          "rules": [
            {
              "required": true,
              "message": "仓库不能为空",
              "trigger": [
                "blur",
                "change"
              ]
            },
            {
              "regex": "",
              "message": "仓库内容不符合规范",
              "trigger": [
                "blur",
                "change"
              ]
            }
          ],
          "visible": {
            "type": "fixed",
            "conditions": [],
            "value": true
          },
          "disabled": false
        },
        "children": [
          {
            "componentName": "d-select",
            "id": 1676461032691,
            "props": {
              "selectType": "normal",
              "singleOptions": {
                "type": "context",
                "fixedOptions": [
                  {
                    "label": "选项",
                    "value": "1",
                    "defaultChecked": false,
                    "visible": false
                  }
                ],
                "apiOptions": {
                  "dataSource": "repoSource",
                  "params": {},
                  "body": {}
                },
                "dataSource": "repoSource"
              },
              "isUpdateChildAll": true,
              "remote": false,
              "allow-clear": false,
              "filter": true,
              "disabled": {
                "type": "fixed",
                "conditions": [],
                "value": false
              }
            },
            "category": "form",
            "children": []
          }
        ]
      },
      {
        "componentName": "d-form-item",
        "id": 1741616266550,
        "category": "form",
        "props": {
          "label": "Continue If Tag Exists",
          "field": "continue_on_existence",
          "help-tips": "Enable this switch to continue the pipeline execution if the tag already exists, otherwise pipeline will fail.",
          "visible": {
            "type": "fixed",
            "conditions": [],
            "value": true
          },
          "disabled": false
        },
        "children": [
          {
            "componentName": "d-switch",
            "id": 1741616266549,
            "props": {
              "defaultValue": false,
              "disabled": {
                "type": "fixed",
                "conditions": [],
                "value": false
              }
            },
            "category": "form"
          }
        ]
      }
    ]
  }
]