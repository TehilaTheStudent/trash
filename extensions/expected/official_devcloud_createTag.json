{
  "plugin_name": "official_devcloud_createTag",
  "display_name": "",
  "data": [
    {
      "unique_id": "b8f7da9c40b24fe88d19cad7b51b42f6",
      "name": "tag_name",
      "default_value": "",
      "plugin_name": "official_devcloud_createTag",
      "version": "0.0.3",
      "type": "d-input",
      "workspace_id": "*",
      "layout_content": "[{\"componentName\":\"d-form\",\"id\":1665991638516,\"category\":\"form\",\"props\":{\"layout\":\"vertical\",\"ref\":\"formRef\"},\"children\":[{\"componentName\":\"d-form-item\",\"id\":1668220990172,\"category\":\"form\",\"props\":{\"label\":\"Tag Name\",\"field\":\"tag_name\",\"help-tips\":\"\",\"rules\":[{\"required\":true,\"message\":\"标签名不能为空\",\"trigger\":[\"blur\",\"change\"]},{\"regex\":\"^.{0,64}$\",\"message\":\"标签名不能超过64个字符\",\"trigger\":[\"blur\",\"change\"]}],\"visible\":{\"type\":\"fixed\",\"conditions\":[],\"value\":true},\"disabled\":false},\"children\":[{\"componentName\":\"d-input\",\"id\":1668220990171,\"props\":{\"inputType\":\"normal\",\"placeholder\":\"请输入\",\"isTrim\":true,\"autocomplete\":\"off\",\"disabled\":{\"type\":\"fixed\",\"conditions\":[],\"value\":false}},\"category\":\"form\"}]},{\"componentName\":\"d-form-item\",\"id\":1676461032692,\"category\":\"form\",\"props\":{\"label\":\"Repository\",\"field\":\"repo_name\",\"help-tips\":\"\",\"rules\":[{\"required\":true,\"message\":\"仓库不能为空\",\"trigger\":[\"blur\",\"change\"]},{\"regex\":\"\",\"message\":\"仓库内容不符合规范\",\"trigger\":[\"blur\",\"change\"]}],\"visible\":{\"type\":\"fixed\",\"conditions\":[],\"value\":true},\"disabled\":false},\"children\":[{\"componentName\":\"d-select\",\"id\":1676461032691,\"props\":{\"selectType\":\"normal\",\"singleOptions\":{\"type\":\"context\",\"fixedOptions\":[{\"label\":\"选项\",\"value\":\"1\",\"defaultChecked\":false,\"visible\":false}],\"apiOptions\":{\"dataSource\":\"repoSource\",\"params\":{},\"body\":{}},\"dataSource\":\"repoSource\"},\"isUpdateChildAll\":true,\"remote\":false,\"allow-clear\":false,\"filter\":true,\"disabled\":{\"type\":\"fixed\",\"conditions\":[],\"value\":false}},\"category\":\"form\",\"children\":[]}]},{\"componentName\":\"d-form-item\",\"id\":1741616266550,\"category\":\"form\",\"props\":{\"label\":\"Continue If Tag Exists\",\"field\":\"continue_on_existence\",\"help-tips\":\"Enable this switch to continue the pipeline execution if the tag already exists, otherwise pipeline will fail.\",\"visible\":{\"type\":\"fixed\",\"conditions\":[],\"value\":true},\"disabled\":false},\"children\":[{\"componentName\":\"d-switch\",\"id\":1741616266549,\"props\":{\"defaultValue\":false,\"disabled\":{\"type\":\"fixed\",\"conditions\":[],\"value\":false}},\"category\":\"form\"}]}]}]",
      "front_data_prop": null,
      "label": null,
      "description": null,
      "required": null,
      "extend_prop": {
        "api_options": null,
        "api_type": null,
        "show_placeholder": null,
        "options": null,
        "disabled_conditions": null,
        "visible_conditions": null
      },
      "validation": {
        "required_message": null,
        "regex": null,
        "regex_message": null,
        "max_length": null,
        "min_length": null
      }
    },
    {
      "unique_id": "0edb313bfee84bf596a07ab29d199716",
      "name": "repo_name",
      "default_value": "",
      "plugin_name": "official_devcloud_createTag",
      "version": "0.0.3",
      "type": "d-select",
      "workspace_id": "*",
      "layout_content": "",
      "front_data_prop": null,
      "label": null,
      "description": null,
      "required": null,
      "extend_prop": {
        "api_options": null,
        "api_type": null,
        "show_placeholder": null,
        "options": null,
        "disabled_conditions": null,
        "visible_conditions": null
      },
      "validation": {
        "required_message": null,
        "regex": null,
        "regex_message": null,
        "max_length": null,
        "min_length": null
      }
    },
    {
      "unique_id": "760ddb3e309b42328c54138bc6e8067e",
      "name": "continue_on_existence",
      "default_value": "false",
      "plugin_name": "official_devcloud_createTag",
      "version": "0.0.3",
      "type": "d-switch",
      "workspace_id": "*",
      "layout_content": "",
      "front_data_prop": null,
      "label": null,
      "description": null,
      "required": null,
      "extend_prop": {
        "api_options": null,
        "api_type": null,
        "show_placeholder": null,
        "options": null,
        "disabled_conditions": null,
        "visible_conditions": null
      },
      "validation": {
        "required_message": null,
        "regex": null,
        "regex_message": null,
        "max_length": null,
        "min_length": null
      }
    }
  ]
}